pkg/shuffler/suite.go

```diff
@@ -66,7 +66,6 @@ func shuffleTests(tests []testing.InternalTest, t *testing.T) []testing.Internal
 		seed = *shuffleSeed
 	}
 	t.Logf("Shuffling tests using seed %d", seed)
```

Comment:
suggestion: if `rand.Seed` was removed due a linter issue, lines 62-68 needs to go, too (as otherwise `seed` isn't actually used).

```diff
-	var seed int64
-	if *shuffleSeed == 0 {
-		seed = time.Now().UnixNano()
-	} else {
-		seed = *shuffleSeed
-	}
-	t.Logf("Shuffling tests using seed %d", seed)
```

Reply:
That's true, I completely missed seeing this change. Yeah, it is technically deprecated, so all this would be unnecessary as well. However, I'm curious if anyone would be relying on deterministic shuffle order for their tests? I would hope not, but unsure if removing would all of a sudden break some tests somewhere. As I'm new to the tooling world, how do we normally handle deprecations? I'm guessing we just follow through with the deprecation, inform teams of the change when the this change gets merged and tagged in master and allow them to upgrade as able?

Reply:
In my experience, if I found an intermittently failing test due to a random seed, I would pin the seed to get it to reliably reproduce and debug. But that was when the test runner supported plumbing in the random seed (Rails).

That being said, despite the fact that this package is [deprecated already](https://github.com/getoutreach/gobox/blob/7dd2d0bb4f3670e58327d917ddc1533ab7a0e6e9/pkg/shuffler/doc.go#L50-L52), there are many internal services which have not migrated off of it (and likely never will unless they are forced to).

If we want to preserve the functionality, we should instead use the recommendation in [the docs for `rand.Seed`](https://pkg.go.dev/math/rand#Seed):

> Programs that call Seed with a known value to get a specific sequence of results should use `New(NewSource(seed))` to obtain a local random generator.